apps/aehyperchains/src/aehc_connector.erl

```diff
@@ -1,36 +1,98 @@
 -module(aehc_connector).
 
--export([broadcast/4, verify/2, subscribe/2]).
+-export([send_tx/1, get_block/1]).
 
--type tx_hash() :: binary().
--type tx_payload() :: binary().
+-export([tx/3, header/2, block/2]).
 
--type account() :: <<_:256>>.
+-export([publish/2, subscribe/1]).
 
 -type connector() :: atom().
 
-%% TODO
--callback broadcast(From::account(), TxPayload::tx_payload(), To::account()) -> 
-    tx_hash().
+-callback send_tx(Tx::aetx:tx()) -> binary().
 
--callback verify(TxHash::tx_hash()) -> 
-    boolean().
+-callback get_block(Num::integer()) -> block().
 
--callback subscribe(Module::atom()) -> ok.
+-export_type([tx/0, header/0, block/0]).
 
--spec broadcast(Con::connector(), From::account(), TxPayload::tx_payload(), To::account()) -> 
-                       tx_hash().
-broadcast(Con, From, TxPayload, To) ->
-    %% Additional params (like value, fee) are parent related and can be passed via appropriate config;
-    Res = Con:broadcast(From, TxPayload, To), true = is_binary(Res),
-    Res.
+-spec connector() -> connector().
+connector() ->
+    ok.
 
--spec verify(Con::connector(), TxHash::tx_hash()) -> 
-                    boolean().
-verify(Con, TxHash) ->
-    Res = Con:verify(TxHash), true = is_boolean(Res), 
-    Res.
+%%%===================================================================
+%%%  Parent chain simplified proto
+%%%===================================================================
 
--spec subscribe(Con::connector(), Module::atom()) -> ok.
-subscribe(Con, Module) ->
-    ok = Con:subscribe(Module).
+-record(tx, {
+             sender :: binary(),
+             recipient :: binary(),
+             payload :: binary()
+            }).
+
+-type tx() :: #tx{}.
+
+-record(header, {
+                 hash :: binary(),
```

Comment:
prev_hash will also be needed to traverse the block structure

Reply:
Important notice, thank you

Reply:
Btw, it could be formatted in a more subtle way
```
-record(name,
    { stuff   :: type()
    , stuffff :: type()
    }
```